docs/src/pages/components/data-grid/components/CustomColumnMenu.tsx

```diff
@@ -0,0 +1,120 @@
+import { makeStyles } from '@material-ui/core/styles';
+import * as React from 'react';
+import Button from '@material-ui/core/Button';
+import {
+  GridColumnMenu,
+  GridColumnMenuContainer,
+  GridColumnMenuProps,
+  GridFilterMenuItem,
+  SortGridMenuItems,
+  XGrid,
+} from '@material-ui/x-grid';
+import StarOutlineIcon from '@material-ui/icons/StarOutline';
+
+const useStyles = makeStyles((theme) => ({
+  primary: {
+    background: theme.palette.primary.main,
+    color: theme.palette.primary.contrastText,
+  },
+  secondary: {
+    background: theme.palette.secondary.main,
+    color: theme.palette.secondary.contrastText,
+  },
+}));
+
+export function CustomColumnMenuComponent(
+  props: GridColumnMenuProps & { color: string },
+) {
+  const classes = useStyles();
+  const { hideMenu, currentColumn, color, ...other } = props;
+
+  if (currentColumn.field === 'name') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <SortGridMenuItems onClick={hideMenu} column={currentColumn!} />
+        <GridFilterMenuItem onClick={hideMenu} column={currentColumn!} />
+      </GridColumnMenuContainer>
+    );
+  }
+  if (currentColumn.field === 'stars') {
+    return (
+      <GridColumnMenuContainer
+        hideMenu={hideMenu}
+        currentColumn={currentColumn}
+        className={classes[color]}
+        {...other}
+      >
+        <div
+          style={{
+            width: 127,
+            height: 160,
+            display: 'flex',
+            justifyContent: 'center',
+            flexDirection: 'column',
+            alignItems: 'center',
+          }}
+        >
+          <StarOutlineIcon style={{ fontSize: 80 }} />
+        </div>
+      </GridColumnMenuContainer>
+    );
+  }
+  return (
+    <GridColumnMenu
+      hideMenu={hideMenu}
+      currentColumn={currentColumn}
+      className={classes[color]}
+      {...other}
+    />
+  );
+}
+
+export default function CustomColumnMenu() {
+  const [color, setColor] = React.useState<'primary' | 'secondary'>('primary');
+  return (
+    <div style={{ height: 300, width: '100%' }}>
+      <Button
+        color={color}
+        onClick={() =>
+          setColor((current) => (current === 'primary' ? 'secondary' : 'primary'))
```

Comment:
Would be nice to also show the column menu the first the user clicks the button. I didn't realize I had to open the menu. 😂